NetKet/Hilbert/hilbert_index.hpp

```diff
@@ -32,7 +32,7 @@ class HilbertIndex {
 
   const int size_;
 
-  std::map<double, int> statenumber_;
+  std::unordered_map<double, int> statenumber_;
```

Comment:
maybe go back to ```std::map``` , I measured a consistent slowing down with ```unordered::map```, but this might be allievated changing the hashing function?

Reply:
> @femtobit , indeed the new version is about 9% slower than the one with std::map on a L=16 spin systems...

Yes, I get the same result now (actually, the difference is even larger on my machine):
```
                  map   N Mean Runtime (s) Std Dev (s)
0  std::unordered_map  14           8,1442  0,0103
2            std::map  14           3,0302  0,0142
1  std::unordered_map  16          47,1230  0,5000
3            std::map  16          16,4941  0,0435
```
For now I can change it back to `std::map`, then.